sway/tree/container.c

```diff
@@ -866,3 +896,72 @@ void container_notify_child_title_changed(struct sway_container *container) {
 size_t container_titlebar_height() {
 	return config->font_height + TITLEBAR_V_PADDING * 2;
 }
+
+void container_set_floating(struct sway_container *container, bool enable) {
+	if (container_is_floating(container) == enable) {
+		return;
+	}
+
+	struct sway_container *workspace = container_parent(container, C_WORKSPACE);
+	struct sway_seat *seat = input_manager_current_seat(input_manager);
+	container_damage_whole(container);
+
+	if (enable) {
+		container_remove_child(container);
+		container_add_child(workspace->sway_workspace->floating, container);
+		if (container->type == C_VIEW) {
+			view_autoconfigure(container->sway_view);
+		}
+		seat_set_focus(seat, seat_get_focus_inactive(seat, container));
+		container_reap_empty_recursive(workspace);
+	} else {
+		// Returning to tiled
+		container_remove_child(container);
+		container_add_child(workspace, container);
+		container->width = container->parent->width;
+		container->height = container->parent->height;
+		if (container->type == C_VIEW) {
+			view_set_maximized(container->sway_view, true);
+		}
+		container->is_sticky = false;
+		container_reap_empty_recursive(workspace->sway_workspace->floating);
+	}
+	arrange_workspace(workspace);
+	container_damage_whole(container);
+}
+
+void container_set_geometry_from_view(struct sway_container *container) {
```

Comment:
The name should make it more obvious that it expects a floating view - maybe `container_set_geometry_from_floating_view`?